lua/starfall/libs_sv/prop.lua

```diff
@@ -472,7 +472,19 @@ function props_library.createSent(pos, ang, class, frozen, data)
 			local value = data[param]
 			
 			if value then
-				checkluatype(value, org[2])
+				-- checkluatype(value, org[2])
+				local typeid = TypeID(value)
+				if typeid ~= org[2] then
+					local typeLookup = {
```

Comment:
I think using SF.GetType here and also modifying the SF.ThrowTypeError to take prefix text would be better than rewriting it all here.

Reply:
Also if we can make a type lookup function SF.TypeName that does this conversion. https://github.com/thegrb93/StarfallEx/blob/master/lua/starfall/sflib.lua#L648

Reply:
Made the changes to sflib a separate pr #923

Reply:
Put it in this PR instead. I don't want to complicate things.